File: code/java/src/main/java/command/ServePastaCommand.java
package command;

class ServePastaCommand implements Command {
    private Receiver receiver;

    ServePastaCommand(Receiver receiver) {
        this.receiver = receiver;
    }

    @Override
    public void execute() {
        System.out.println("ServePastaCommand.execute(): Delegating to Receiver.servePasta()");
        receiver.servePasta();
    }
}

File: code/typescript/src/command/ServePastaCommand.ts
import { Command } from "./Command";
import { Receiver } from "./Receiver";

export class ServePastaCommand implements Command {
    private receiver: Receiver;

    constructor(receiver: Receiver) {
        this.receiver = receiver;
    }

    execute(): void {
        console.log("ServePastaCommand.execute(): Delegating to Receiver.servePasta()");
        this.receiver.servePasta();
    }
}
